internal/httpclient/http.go

```diff
@@ -0,0 +1,102 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	urlpkg "net/url"
+)
+
+// HTTPClient is a convenience wrapper for http.Client that consolidates common logic.
+type HTTPClient interface {
+	// Get returns the body and media type of the URL using the provided context. The caller must close the body.
+	//
+	// This is optimized for easy content negotiation. Hence, the returned mediaType is stripped of qualifiers.
+	// Ex. "Content-Type: application/json; charset=utf-8" will return mediaType "application/json"
+	Get(ctx context.Context, url string, header *http.Header) (body io.ReadCloser, mediaType string, err error)
+	// GetJSON is a convenience function that calls json.Unmarshal after Get.
+	GetJSON(ctx context.Context, url string, accept string, v interface{}) error
+}
+
+type httpClient struct{ client http.Client }
+
+// New returns a client that implicitly authenticates when it needs to
+// Use ContextWithClientTransport when testing.
+func New(transport http.RoundTripper) HTTPClient {
+	return &httpClient{client: http.Client{Transport: transport}}
+}
+
+type contextClientTransportKey struct{}
+
+// TransportFromContext returns an http.RoundTripper from the context or nil
+func TransportFromContext(ctx context.Context) http.RoundTripper {
+	if v, ok := ctx.Value(contextClientTransportKey{}).(http.RoundTripper); ok {
+		return v
+	}
+	return http.DefaultTransport
+}
+
+// ContextWithTransport returns a context with a http.RoundTripper.
+func ContextWithTransport(ctx context.Context, transport http.RoundTripper) context.Context {
+	return context.WithValue(ctx, contextClientTransportKey{}, transport)
+}
+
+func (h *httpClient) Get(ctx context.Context, url string, header *http.Header) (io.ReadCloser, string, error) {
+	u, err := urlpkg.Parse(url)
+	if err != nil {
+		return nil, "", err
+	}
+
+	hdr := http.Header{}
+	if *header != nil {
+		hdr = header.Clone()
+	}
+	hdr.Add("User-Agent", "car/dev")
+	req := &http.Request{Method: "GET", URL: u, Header: hdr}
+	res, err := h.client.Do(req.WithContext(ctx))
+	if err != nil {
+		return nil, "", err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		res.Body.Close() //nolint
+		return nil, "", fmt.Errorf("received %v status code from %q", res.StatusCode, url)
+	}
+
+	contentType := res.Header.Get("Content-Type")
+	mediaType, _, _ := mime.ParseMediaType(contentType) // strip qualifiers
+	return res.Body, mediaType, nil
+}
+
+func (h *httpClient) GetJSON(ctx context.Context, url, accept string, v interface{}) error {
+	body, _, err := h.Get(ctx, url, &http.Header{"Accept": []string{accept}})
```

Comment:
I believe better use `.Add` rather than giving initial value directly in order to have consistent case of characters

Reply:
sgtm I think I was overly trying to save a line :D